### PR TITLE
Added support to request V3 tokens

### PR DESCRIPTION
**Pull request by mikezuniga**
Adding support to request V3 tokens and provision compute using v3 keystone style auth

### CODE REVIEW

**Comment by mikezuniga** [comment]
Are we going to be able to merge this? Seems there are some people interested on this.

File: lib/fog/openstack/core.rb [hunk 1]
```diff
@@ -169,13 +173,113 @@ def self.authenticate_v2(options, connection_options = {})
...
+    # Keystone Style Auth
+    def self.authenticate_v3(options, connection_options = {})
+      uri                   = options[:openstack_auth_uri]
+      tenant_name           = options[:openstack_tenant]
+      service_type          = options[:openstack_service_type]
+      service_name          = options[:openstack_service_name]
+      identity_service_type = options[:openstack_identity_service_type]
+      endpoint_type         = (options[:openstack_endpoint_type] || 'public').to_s
+      openstack_region      = options[:openstack_region]
+      domain_name           = options[:openstack_domain]
```
Comment (TerryHowe) [inline lib/fog/openstack/core.rb]: Just wanted to note again that user and project domains will probably needed at least a future enhancement

**Comment by geemus** [comment]
Thanks!

@TerryHowe good point, sounds like it wasn't a prereq, so I brought this in, would be great to see that enhancement though.

**Comment by mikezuniga** [comment]
I can work on it during the weekend, it wont be much of a change, I only have to pass the parameters and build the hash that is sent as json based on which params are available. 

I'll put in a reminder on my calendar :)

**Comment by geemus** [comment]
Great, thanks!

**Pull request merged**